worker/deployer/manifold.go

```diff
@@ -41,35 +39,9 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
 func (config ManifoldConfig) newWorker(a agent.Agent, apiCaller base.APICaller) (worker.Worker, error) {
 	cfg := a.CurrentConfig()
 	// Grab the tag and ensure that it's for a machine.
-	tag, ok := cfg.Tag().(names.MachineTag)
-	if !ok {
+	if cfg.Tag().Kind() != names.MachineTagKind {
 		return nil, errors.New("agent's tag is not a machine tag")
 	}
-
-	// Get the machine agent's jobs.
-	// TODO(fwereade): this functionality should be on the
-	// deployer facade instead.
-	agentFacade, err := apiagent.NewState(apiCaller)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	entity, err := agentFacade.Entity(tag)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	var isUnitHoster bool
-	for _, job := range entity.Jobs() {
-		if job == multiwatcher.JobHostUnits {
-			isUnitHoster = true
-			break
-		}
-	}
-
-	if !isUnitHoster {
-		return nil, dependency.ErrUninstall
```

Comment:
should we still do this check?

Reply:
All machines are created with JobHostUnits. A long time ago maybe that wasn't the case. This check is only relevant if machines are run as JobManageModel only which is never the case